packages/core/src/services/commits/promptCache.ts

```diff
@@ -0,0 +1,98 @@
+import { hash } from 'crypto'
+
+import { Config, Conversation } from '@latitude-data/compiler'
+
+import { ChainStepResponse, StreamType, type Workspace } from '../../browser'
+import { cache } from '../../cache'
+
+class DocumentCache {
+  private workspace: Workspace
+  private config: Config
+  private conversation: Conversation
+
+  constructor({
+    workspace,
+    config,
+    conversation,
+  }: {
+    workspace: Workspace
+    config: Config
+    conversation: Conversation
+  }) {
+    this.workspace = workspace
+    this.config = config
+    this.conversation = conversation
+  }
+
+  private async getCache() {
+    return await cache()
+  }
+
+  private generateKey() {
+    const k = hash(
+      'sha256',
+      `${JSON.stringify(this.conversation)}:${JSON.stringify(this.config)}`,
+    )
+
+    return `workspace:${this.workspace.id}:prompt:${k}`
+  }
+
+  private shouldCache(): boolean {
+    const temp = parseFloat(this.config.temperature as any)
+    return isNaN(temp) || temp === 0
+  }
+
+  async get(): Promise<ChainStepResponse<StreamType> | undefined> {
+    try {
+      if (!this.shouldCache()) return undefined
+
+      const c = await this.getCache()
+      const key = this.generateKey()
+      const cachedResponseStr = await c.get(key)
+
+      return cachedResponseStr ? JSON.parse(cachedResponseStr) : undefined
+    } catch (e) {
+      return undefined
+    }
+  }
+
+  async set(response: ChainStepResponse<StreamType>): Promise<void> {
+    try {
+      if (!this.shouldCache()) return
+
+      const c = await this.getCache()
+      const key = this.generateKey()
+      await c.set(key, JSON.stringify(response))
+    } catch (e) {
+      // Silently fail cache writes
+    }
+  }
+}
+
+export async function getCachedResponse({
+  workspace,
+  config,
+  conversation,
+}: {
+  workspace: Workspace
+  config: Config
+  conversation: Conversation
+}) {
+  const documentCache = new DocumentCache({ workspace, config, conversation })
+  return await documentCache.get()
+}
+
+export async function setCachedResponse({
+  workspace,
+  config,
+  conversation,
+  response,
+}: {
+  workspace: Workspace
+  config: Config
+  conversation: Conversation
+  response: ChainStepResponse<StreamType>
+}) {
+  const documentCache = new DocumentCache({ workspace, config, conversation })
```

Comment:
Why a class in this case?

Reply:
yeah no real reason to be honest it's what the ai generated lol, I will change it to pure functions